src/include/daos/rpc.h

```diff
@@ -64,6 +66,68 @@ enum daos_module_id {
 
 };
 
+#define DAOS_VOS_VERSION      1
+#define DAOS_MGMT_VERSION     4
+#define DAOS_POOL_VERSION     7
+#define DAOS_CONT_VERSION     8
+#define DAOS_OBJ_VERSION      10
+#define DAOS_REBUILD_VERSION  4
+#define DAOS_RSVC_VERSION     4
+#define DAOS_RDB_VERSION      4
+#define DAOS_RDBT_VERSION     3
+#define DAOS_SEC_VERSION      1
+#define DAOS_DTX_VERSION      4
+#define DAOS_PIPELINE_VERSION 1
+#define DAOS_CHK_VERSION      1
+
+#define DAOS_MAX_PROTOCOLS    2
+struct daos_protocol_table {
+	uint8_t protocol;
+	uint8_t versions[DAOS_NR_MODULE];
+};
+
+static const struct daos_protocol_table daos_rpc_protocol_tables[] = {
+    {
+	/* Latest protocol */
+	.protocol = DAOS_VERSION_PROTOCAL,
+	.versions = {DAOS_VOS_VERSION, DAOS_MGMT_VERSION, DAOS_POOL_VERSION, DAOS_CONT_VERSION,
+		     DAOS_OBJ_VERSION, DAOS_REBUILD_VERSION, DAOS_RSVC_VERSION, DAOS_RDB_VERSION,
+		     DAOS_RDBT_VERSION, DAOS_SEC_VERSION, DAOS_DTX_VERSION, DAOS_PIPELINE_VERSION,
+		     DAOS_CHK_VERSION},
+    },
+    /* Please update DAOS_VERSION_PROTOCOL - 1 table when rolling upgrade is supported.
+    {
+    .protocol = DAOS_VERSION_PROTOCAL - 1;
+    .versions = {DAOS_VOS_VERSION, DAOS_MGMT_VERSION, DAOS_POOL_VERSION, DAOS_CONT_VERSION,
+		 DAOS_OBJ_VERSION, DAOS_REBUILD_VERSION, DAOS_RSVC_VERSION, DAOS_RDB_VERSION,
+		 DAOS_RDBT_VERSION, DAOS_SEC_VERSION, DAOS_DTX_VERSION, DAOS_PIPELINE_VERSION,
+		 DAOS_CHK_VERSION},
+    },
+    */
+};
+
+static inline int
+daos_get_module_version(int module_id, uint8_t protocol, uint8_t *version)
```

Comment:
Sounds like it's getting RPC version by 'module' and 'protocol'? daos_get_rpc_version() might be a better name.

Reply:
Yup, looks better.